pkg/trace/traceutil/azure.go

```diff
@@ -71,27 +74,83 @@ func getAppServicesTags(getenv func(string) string) map[string]string {
 		aasSiteType:        appService,
 		aasSubscriptionID:  subscriptionID,
 	}
+
+	// Remove the Java and .NET logic once non-universal extensions are deprecated
+	if websiteOS == "windows" {
+		if len(extensionVersion) > 0 {
+			tags[aasExtensionVersion] = extensionVersion
+		} else if hasEnv("DD_AAS_JAVA_EXTENSION_VERSION", getenv) {
+			tags[aasExtensionVersion] = getenv("DD_AAS_JAVA_EXTENSION_VERSION")
+		} else if hasEnv("DD_AAS_DOTNET_EXTENSION_VERSION", getenv) {
+			tags[aasExtensionVersion] = getenv("DD_AAS_DOTNET_EXTENSION_VERSION")
+		}
```

Comment:
Using `hasEnv` poses a bit of challenge here, since we have to pass in `getenv` as a parameter but we don't actually get the env value anyways! We should be able to do something cleaner like:

```go
...
} else if val := getenv("DD_AAS_JAVA_EXTENSION_VERSION"); val != "" {
	tags[aasExtensionVersion] = val
} else if val := getenv("DD_AAS_DOTNET_EXTENSION_VERSION"); val != "" {
	tags[aasExtensionVersion] = val
}
```

Reply:
Didn't know this was possible in Go, thanks!